coffee_lib/src/plugin_manager.rs

```diff
@@ -40,4 +40,7 @@ pub trait PluginManager {
 
     /// show the README file of the pulgin
     async fn show(&mut self, plugin: &str) -> Result<Value, CoffeeError>;
+
+    /// search the remote repositories for the plugin
+    async fn search(&mut self, plugin: &str) -> Result<Value, CoffeeError>;
```

Comment:
Ok, wait a minute here, now I was thinking what is the meaning of this `coffee seach` command?

What is the meaning of knowing the remote if the user adds the remote after all?

Reply:
In my opinion, the purpose of this command would be to make sure whether the plugin is present in the added remote repositories or not, which can be useful prior to installation. 

Nonetheless, after further consideration, it appears that the `coffee show` could fulfill the same objective in a different way.

Reply:
I really do not know if this command is the right things to add, reckless does this because it has an internal repository, but we use the remote, so you should know already that this plugin is already in the repository.

Lets leave this here for a while and work on other commands that we needs more than this, like removing a plugin